tools/cythonize.py

```diff
@@ -55,8 +56,25 @@ def process_pyx(fromfile, tofile, cwd):
     try:
         from Cython.Compiler.Version import version as cython_version
         from distutils.version import LooseVersion
-        if LooseVersion(cython_version) < LooseVersion('0.23.4'):
-            raise Exception('Building SciPy requires Cython >= 0.23.4')
+
+        # Try to find pyproject.toml
+        pyproject_toml = os.path.join(dirname(dirname(__file__)), "pyproject.toml")
```

Comment:
Ha yes that's a more elegant way to solve the problem.

Reply:
It works!

```
Building, see build.log...
Running scipy/linalg/_generate_pyx.py
Running scipy/special/_generate_pyx.py
Processing scipy/_lib/_ccallback_c.pyx
Processing scipy/_lib/messagestream.pyx
Processing scipy/cluster/_hierarchy.pyx
Traceback (most recent call last):
  File "/home/ubuntu/scipy/tools/cythonize.py", line 283, in <module>
    main()
  File "/home/ubuntu/scipy/tools/cythonize.py", line 279, in main
    find_process_files(root_dir)
  File "/home/ubuntu/scipy/tools/cythonize.py", line 268, in find_process_files
    for result in pool.imap(lambda args: process(*args), jobs):
  File "/opt/circleci/python/2.7.11/lib/python2.7/multiprocessing/pool.py", line 668, in next
    raise value
Exception: Building SciPy requires Cython >= 0.25
Processing scipy/cluster/_optimal_leaf_ordering.pyx
Processing scipy/cluster/_vq.pyx
Processing scipy/interpolate/_bspl.pyx
Processing scipy/interpolate/_ppoly.pyx
Processing scipy/interpolate/interpnd.pyx
Processing scipy/io/matlab/mio5_utils.pyx
Processing scipy/io/matlab/mio_utils.pyx
Processing scipy/io/matlab/streams.pyx
Processing scipy/linalg/_decomp_update.pyx.in
```

Reply:
I'd prefer `join(dirname(__file__), '..', '..', 'pyproject.toml')` over the iterated dirname.

Reply:
I think you have an extra `..` in there, but will do.